projects/adf4377_sdz/src/app/adf4377_sdz.c

```diff
@@ -0,0 +1,145 @@
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+/******************************************************************************/
+/***************************** Include Files **********************************/
+/******************************************************************************/
+#include <xparameters.h>
+#include "spi.h"
+#include "spi_extra.h"
+#include "gpio.h"
+#include "gpio_extra.h"
+#include "error.h"
+#include "adf4377.h"
+#include "parameters.h"
+#include "print_log.h"
+#include "util.h"
+
+#ifdef IIO_SUPPORT
+#include "iio_app.h"
+#include "iio_adf4377.h"
+#endif
+
+int main(void)
+{
+	int32_t ret;
+	uint8_t f_lock;
+	struct adf4377_dev *dev;
+
+	struct xil_spi_init_param xil_spi_init = {
+		.flags = 0,
+		.type = SPI_PS
+	};
+
+	struct xil_gpio_init_param xil_gpio_init = {
+		.device_id = GPIO_DEVICE_ID,
+		.type = GPIO_PS
+	};
+
+	struct gpio_init_param gpio_ce_param = {
+		.number = GPIO_CE,
+		.platform_ops = &xil_gpio_platform_ops,
+		.extra = &xil_gpio_init
+	};
+
+	struct gpio_init_param gpio_enclk1_param = {
+		.number = GPIO_ENCLK1,
+		.platform_ops = &xil_gpio_platform_ops,
+		.extra = &xil_gpio_init
+	};
+
+	struct gpio_init_param gpio_enclk2_param = {
+		.number = GPIO_ENCLK2,
+		.platform_ops = &xil_gpio_platform_ops,
+		.extra = &xil_gpio_init
+	};
+
+	struct spi_init_param spi_init = {
+		.device_id = SPI_DEVICE_ID,
+		.max_speed_hz = 2000000,
+		.chip_select = SPI_ADF4377_CS,
+		.mode = SPI_MODE_0,
+		.bit_order = SPI_BIT_ORDER_MSB_FIRST,
+		.platform_ops = &xil_platform_ops,
+		.extra = &xil_spi_init
+	};
+
+	struct adf4377_init_param adf4377_param = {
+		.dev_id = ADF4377,
+		.spi_init = &spi_init,
+		.gpio_ce_param = &gpio_ce_param,
+		.gpio_enclk1_param = &gpio_enclk1_param,
+		.gpio_enclk2_param = &gpio_enclk2_param,
+		.spi4wire = true,
+		.clkin_freq = 100000000,
+		.cp_i = ADF4377_CP_10MA1,
+		.muxout_select = ADF4377_MUXOUT_HIGH_Z,
+		.ref_doubler_en = ADF4377_REF_DBLR_DIS,
+		.f_clk = 1000000000,
+		.clkout_op = ADF4377_CLKOUT_427MV
+	};
+
+	ret = adf4377_init(&dev, &adf4377_param);
+	if (ret != SUCCESS) {
+		pr_err("ADF4377 Initialization failed!\n");
+		return FAILURE;
+	}
+
+	pr_info("ADF4377 Successfully initialized!");
+
+	ret = adf4377_spi_read(dev, ADF4377_REG(0x49), &f_lock);
+	if (ret != SUCCESS) {
+		pr_err("SPI Read Failed!");
+	}
+
+	if(ADF4377_LOCKED(f_lock))
+		pr_info("Output Frequency Locked!");
+	else
+		pr_err("Output Frequency not Locked!");
```

Comment:
One is info and one is error?

Reply:
yes. pr_info is used to inform the user that the output was generated successfully.

As for the error:
"pr_err - An error condition, often used by drivers to indicate difficulties with the hardware" (Reference: https://elinux.org/Debugging_by_printing)